ndless-sdk/toolchain/build_toolchain.sh

```diff
@@ -49,45 +56,117 @@ if ! gcc -lgmp       test.c -o test; then error=1; echo 'GMP (gmp-devel/libgmp-d
 if ! gcc -lmpfr      test.c -o test; then error=1; echo 'MPFR (mpfr-devel/libmpfr-dev) dependency seems to be missing!'; fi
 if ! gcc -lmpc       test.c -o test; then error=1; echo 'MPC (mpc-devel/libmpc-dev) dependency seems to be missing!'; fi
 if ! gcc -lz         test.c -o test; then error=1; echo 'zlib (zlib-devel/zlib1g-dev) dependency seems to be missing!'; fi
-if ! gcc -lpython2.7 test.c -o test; then error=1; echo 'libpython2.7 (python-devel/python2.7-dev) dependency seems to be missing!'; fi
+if [ -n "${PYTHON}" ] && [ ! -x "${PYTHON}-config" ]; then error=1; echo "${PYTHON}-config (python3-devel/python3-dev) dependency seems to be missing!"; fi
 rm -f test test.c
 [ $error -eq 1 ] && exit 1
 
 mkdir -p build download
 
-if [ ! -f .downloaded ]; then
-	wget -c http://ftp.gnu.org/gnu/binutils/${BINUTILS}.tar.bz2 -O download/${BINUTILS}.tar.bz2 && tar xvjf download/${BINUTILS}.tar.bz2 && \
-	wget -c ftp://ftp.gnu.org/gnu/gcc/${GCC}/${GCC}.tar.xz      -O download/${GCC}.tar.xz       && tar xvJf download/${GCC}.tar.xz && \
-	wget -c ftp://sourceware.org/pub/newlib/${NEWLIB}.tar.gz    -O download/${NEWLIB}.tar.gz    && tar xvzf download/${NEWLIB}.tar.gz && \
-	wget -c ftp://ftp.gnu.org/gnu/gdb/${GDB}.tar.xz             -O download/${GDB}.tar.xz       && tar xvJf download/${GDB}.tar.xz && \
-	touch .downloaded
-	if [ $? -ne 0 ]; then
-		echo "Download failed!"
-		exit 1
+downloadAndExtract() {
+	url="$1"
+	target="download/$(basename "${url}")"
+	if ! wget -c "$url" -O "${target}"; then
+		rm -f "${target}"
+		return 1
 	fi
-fi
+	tar -xaf "${target}" --checkpoint=.10 -C download
```

Comment:
`--checkpoint` doesn't exist on macOS/[BSD tar](https://www.freebsd.org/cgi/man.cgi?tar(1)).
If we really want it, we could probably pipe to `pv` etc. but meh?

Reply:
Is there some equivalent option?

Reply:
Well, I see alternatives proposed here: https://superuser.com/questions/168749/is-there-a-way-to-see-any-tar-progress-per-file

Reply:
Let's just drop this, then. I added an echo so that it doesn't seem stuck.